features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/source/src/app/components/Apis/Details/Configuration/ApiSecurity.jsx

```diff
@@ -0,0 +1,495 @@
+        const isMandatory = value === 'Mandatory';
+
+        if (name === 'mutualSSLMandatoryState') {
+            if (isMandatory) addToArray(securitySchemaValues.mutualSSLMandatory, tempSecurityScheme);
+            else {
+                removeFromArray(securitySchemaValues.mutualSSLMandatory, tempSecurityScheme);
+                if (tempSecurityScheme.includes(securitySchemaValues.oauth2) ||
+                tempSecurityScheme.includes(securitySchemaValues.basicAuth)
+                ) {
+                    addToArray(securitySchemaValues.oauthBasicAuthMandatory, tempSecurityScheme);
+                }
+            }
+        } else if (name === 'oauthBasicAuthMandatoryState') {
+            if (isMandatory) addToArray(securitySchemaValues.oauthBasicAuthMandatory, tempSecurityScheme);
+            else {
+                removeFromArray(securitySchemaValues.oauthBasicAuthMandatory, tempSecurityScheme);
+                if (tempSecurityScheme.includes(securitySchemaValues.mutualSSL)) {
+                    addToArray(securitySchemaValues.mutualSSLMandatory, tempSecurityScheme);
+                }
+            }
+        }
+        setSecurityScheme(tempSecurityScheme);
+    };
+
+    const showMandatoryOption = (apiSecurityScheme) => {
+        let tempSecurityScheme = securityScheme;
+        if (!tempSecurityScheme) tempSecurityScheme = apiSecurityScheme;
+
+        const mutualSSLWithOauthOrBasicAuth = tempSecurityScheme.includes(securitySchemaValues.mutualSSL) &&
+        (tempSecurityScheme.includes(securitySchemaValues.oauth2) ||
+        tempSecurityScheme.includes(securitySchemaValues.basicAuth));
+
+        return mutualSSLWithOauthOrBasicAuth;
+    };
+
+    const getAPISecurityMessage = (apiSecurityScheme) => {
+        let tempSecurityScheme = securityScheme;
+        if (!tempSecurityScheme) tempSecurityScheme = apiSecurityScheme;
+        if (tempSecurityScheme.includes(securitySchemaValues.mutualSSLMandatory) &&
+        tempSecurityScheme.includes(securitySchemaValues.oauthBasicAuthMandatory)) {
+            return 'Transport security and Authentication both are set as mandatory';
+        } else if (tempSecurityScheme.includes(securitySchemaValues.mutualSSLMandatory) &&
+        !tempSecurityScheme.includes(securitySchemaValues.oauthBasicAuthMandatory)) {
+            return 'Transport security set as mandatory and Authentication are set as optional';
+        } else {
+            return 'Transport security set as optional and Authentication are set as mandatory';
+        }
+    };
+
+    return (
+        <div>
+            <Grid
+                container
+                direction='row'
+                alignItems='flex-start'
+                spacing={16}
+                className={classes.buttonSection}
+            >
+                <Grid item style={{ width: '250px' }}>
+                    {/* Transport Security */}
+                    {isTransportHttps && (
```

Comment:
need space in after opening curly bracer and before closing curly bracer. Fix else where too

Reply:
Fixed in https://github.com/wso2/carbon-apimgt/commit/636f93611578bba36bc677fb4eb1bcdf281489a5